### PR TITLE
add tare force torque sensor service and skill

### PR DESCRIPTION
**Pull request by JohnTGZ**
This PR:

1. Adds a Tare F/T sensor ROS service server to the `robot_control_bridge` plugin. 
  - Service is similar to the  `aic_controller/tare_force_torque_sensor` service from `aic_controller`. 

2. Adds a `tare_force_torque_sensor_skill` Flowstate skill
  - This skill calls the `aic_controller/tare_force_torque_sensor` service in  `robot_control_bridge`  to trigger taring of the F/T sensor. 
    - This results in the F/T sensor readings being zero-ed.
    - The service callback will re-establish a connection to the robot controller if it is not already connected.

### CODE REVIEW

File: flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp [hunk 1]
```diff
@@ -515,6 +544,49 @@ void RobotControlBridge::RestartBridgeCallback(
...
+  if (!data_->connected_to_controller_) {
+    for (int i = 0; i < data_->restart_connection_retries_; ++i) {
+      data_->connected_to_controller_ = restartControllerBridge();
+      if (data_->connected_to_controller_) {
+        break;
+      }
+      if (i < data_->restart_connection_retries_ - 1) {
+        LOG(INFO)
+            << "Failed to restart controller bridge. Retrying in 500ms...";
+        rclcpp::sleep_for(std::chrono::milliseconds(500));
```
Comment (kaushikbalasundar) [inline flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp]: This node uses the `EventsExecutor`, which operates as a single-threaded executor. When you use `rclcpp::sleep_for(500)` in your callback, it will block the entire node for that duration, preventing any other topics or services handled by this bridge from processing during that time (maximum of 1.5s -> 500ms x 3 re-tries). Consider using a non-blocking retry mechanism ? 

One way I can think of is using a background thread that doesn't block the main ROS thread:

`std::thread([this]() {
  for (int i = 0; i < data_->restart_connection_retries_; ++i) {
    if (restartControllerBridge()) break;
    std::this_thread::sleep_for(std::chrono::milliseconds(500));
  }
}).detach();`

File: flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp [hunk 2]
```diff
@@ -692,8 +783,13 @@ bool RobotControlBridge::startControllerSession() {
   data_->num_joints_ = part_status.value().joint_states_size();
 
   // Start ICON Session
-  auto session_or =
-      Session::Start(icon_channel_or.value(), {data_->part_name_});
+  std::vector<std::string> parts = {data_->part_name_};
+  if (data_->ft_sensor_part_name_ != data_->part_name_) {
+    parts.push_back(data_->ft_sensor_part_name_);
+  }
+  // todo(johntgz) test if starting a session with multiple parts would cause an
```
Comment (kaushikbalasundar) [inline flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp]: Is this verified? The controller doesn't seem to complain - is that sufficient to call it verified?

**Comment by kaushikbalasundar** [comment]
A couple of comments - LGTM otherwise.

File: flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp (hunk 1 above)
Comment (JohnTGZ) [reply flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp]: The sleep is waiting for the robot controller to transition to ENABLED state after a restart.
I did try to run a detached background thread but got mixed results as the other callbacks in this plugin depend on having established a connection to the robot controller.  

I settled on blocking the current thread so everything can happen in sequence: 
Establish connection to the robot controller -> Tare the force torque sensor -> Return a successful service response -> (Thread runs the callback queue for motion commands).

The side-effect is that we also block the other plugins in the bridge when a request to tare F/T sensor is received. 
What I can do right now is to increase the number of retries to `6` and reduce the sleep duration to `250 ms` so as to potentially reduce the blocking duration.
And I will also look at doing further tests and figuring out what is the issue behind using background threads.

Comment (JohnTGZ) [reply flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp]: Updated sleep duration and number of retries in [5ac3ee2](https://github.com/intrinsic-dev/aic/pull/510/commits/5ac3ee2484c827efe81a804aae90d83fe1be8b60)

File: flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp (hunk 2 above)
Comment (JohnTGZ) [reply flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp]: Yes it has been! Removed in [5ac3ee2](https://github.com/intrinsic-dev/aic/pull/510/commits/5ac3ee2484c827efe81a804aae90d83fe1be8b60)

File: flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp (hunk 1 above)
Comment (kaushikbalasundar) [reply flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp]: Given this isn't causing any visible issues for the timebeing, and based on offline conversation, I'm okay noting this down as future improvement and merging it for now. Perhaps let's a note here to come back to this?

**Review by kaushikbalasundar — APPROVED** [review]

File: flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp (hunk 1 above)
Comment (JohnTGZ) [reply flowstate/aic_flowstate_ros_bridge/src/robot_control_bridge.cpp]: Got it. I have added a TODO comment for that.
